3.Linear_Modeling/nbconverted/fit_linear_model.py

```diff
@@ -0,0 +1,364 @@
+        + "_"
+        + feature_df["Metadata_dose"]
+    )
+    .dropna()
+    .reset_index()
+)
+
+cp_features = infer_cp_features(feature_df)
+print(f"We are testing {len(cp_features)} CellProfiler features")
+
+feature_df["Metadata_Treatment_and_Dose"].unique()
+
+
+# ##### Here I plot the beta coefficients for each treatment against the number of cells per well. Data points the drift heavily in the Y axis are features that are affected the most by the y-axis treatment while data points that drift more in the x-axis are features that are most affected by the number of cells in a well.
+
+# #### Simple Linear Modeling (cell count beta + 1 beta approach)
+# Here I merged the treatment and dosage and used DMSO 0.1% as the control simply comparing one dosage/treatment at a time and outputting each graph for each treatment for all features. All features and treatments will be exported into 1 file.
+#
+# Linear Model:
+# $y = \beta _{0}x+ \beta _{1}x+ \epsilon$ where;
+# $y$ is each feature
+# $x$ is the inputed variables
+# $\beta _{0}$ is the beta coefficient attributed to cell count,
+# $\beta _{1}$ is the beta coefficient attributed to treatment & dose combined.
+# $\epsilon$ is the residual variance not explained by factors in the model
+
+# In[9]:
+
+
+control = ["DMSO 0.1%_0"]
+# lm_results_df_all = pd.DataFrame(cp_features, columns=["feature"])
+# lm_results_df_all
+lm_results = []
+for i in feature_df["Metadata_Treatment_and_Dose"].unique():
+    treatment = []
+    treatment.append(i)
+
+    dosage_treatments_list = treatment + control
+    print(dosage_treatments_list)
+    df = feature_df.query("Metadata_Treatment_and_Dose == @dosage_treatments_list")
+
+    df["Metadata_Treatment_and_Dose"] = LabelEncoder().fit_transform(
+        df["Metadata_Treatment_and_Dose"]
+    )
+
+    # Setup linear modeling framework
+    variables = ["Metadata_number_of_singlecells"]
+    X = df.loc[:, variables]
+    X = pd.concat([X, df["Metadata_Treatment_and_Dose"]], axis=1)
+
+    columns_list = (
+        ["feature", "r2_score"] + X.columns.tolist() + ["dosage_treatments_list"]
+    )
+    # Fit linear model for each feature
+    for cp_feature in cp_features:
+        # Subset CP data to each individual feature (univariate test)
+        cp_subset_df = df.loc[:, cp_feature]
+
+        # Fit linear model
+        lm = LinearRegression(fit_intercept=True)
+        lm_result = lm.fit(X=X, y=cp_subset_df)
+
+        # Extract Beta coefficients(contribution of feature to X covariates)
+        coef = list(lm_result.coef_)
+        # Estimate fit (R^2)
+        r2_score = lm.score(X=X, y=cp_subset_df)
+
+        # Add results to a growing list
+        lm_results.append(
+            [cp_feature, r2_score] + coef + [f"{'_'.join(dosage_treatments_list)}"]
+        )
+
+# Convert results to a pandas DataFrame
+lm_results_df = pd.DataFrame(lm_results, columns=columns_list)
+lm_results_df.head(100)
+
+
+# define output file path
```

Comment:
traditionally, we define all inputs and outputs at the beginning of the notebook (after imports and function definitions). This way there are no surprises about what the notebook is ingesting and outputting

Reply:
set main paths to beginning of the notebook, but needed to update file path specific names from variables defines later in the notebook.